src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -150,15 +150,18 @@ internal static bool IsLegalSpanStackAllocPosition(this SyntaxNode node)
 
             switch (parentNode.Kind())
             {
+                // In case of a declaration of a Span<T> variable
                 case SyntaxKind.EqualsValueClause:
                     {
-                        // In case of a declaration of a Span<T> variable
-                        return parentNode.Parent.IsKind(SyntaxKind.VariableDeclarator);
+                        SyntaxNode variableDeclarator = parentNode.Parent;
+
+                        return variableDeclarator.IsKind(SyntaxKind.VariableDeclarator) &&
+                            variableDeclarator.Parent.IsKind(SyntaxKind.VariableDeclaration);
                     }
+                // In case of reassignment to a Span<T> variable
                 case SyntaxKind.SimpleAssignmentExpression:
```

Comment:
How about in the case of calling a method that accepts a Span, passing in a stackalloc? Is that allowed?